operations/node_management/Add_Remove_Replace_NCNs/Remove_NCN_from_Role.md

```diff
@@ -0,0 +1,494 @@
+            --key=/etc/kubernetes/pki/etcd/ca.key --endpoints=localhost:2379 member list
+    ```
+
+1. Remove the master node from the etcd cluster backing Kubernetes.
+
+    Replace the `<MEMBER_ID>` value with the value returned in the previous sub-step.
+
+    ```bash
+    ncn-m# etcdctl --cacert=/etc/kubernetes/pki/etcd/ca.crt --cert=/etc/kubernetes/pki/etcd/ca.crt \
+            --key=/etc/kubernetes/pki/etcd/ca.key --endpoints=localhost:2379 member remove <MEMBER_ID>
+    ```
+
+#### Stop kubelet, containerd and Etcd services ***on the master node being removed***
+
+```bash
+ncn-m# systemctl stop kubelet.service ; systemctl stop containerd.service ; systemctl stop etcd.service
+```
+
+#### Add the node back into the etcd cluster
+
+This will allow the node to rejoin the cluster automatically when it gets added back.
+
+- The IP address and hostname of the rebuilt node is needed for the following command.
+- Replace the `<IP_ADDRESS>` address value with the IP address you noted in an earlier step from the `etcdctl` command.
+
+```bash
+ncn-mw# etcdctl --cacert=/etc/kubernetes/pki/etcd/ca.crt --cert=/etc/kubernetes/pki/etcd/ca.crt \
+            --key=/etc/kubernetes/pki/etcd/ca.key --endpoints=localhost:2379 member add $NODE \
+            --peer-urls=https://<IP_ADDRESS>:2380
+```
+
+#### Remove Etcd data directory ***on the master node being removed***
+
+```bash
+ncn-m# rm -rf /var/lib/etcd/*
+```
+
+#### Save a copy of the `lan0` config from `ncn-m001` **only if `ncn-m001` is being removed**
+
+**Skip this step if `ncn-m001` is not being removed.**
+
+```bash
+ncn-m001# rsync /etc/sysconfig/network/ifcfg-lan0 ncn-m002:/tmp/ifcfg-lan0-m001
+```
+
+#### Master node role removal complete
+
+The master node role removal is complete; proceed to [wipe the drives](#wipe-disks-master-node).
+
+<a name="worker-node-remove-roles"></a>
+
+### Worker Node Remove Roles
+
+#### Drain the node to clear any pods running on the node
+
+**IMPORTANT:** The following command will cordon and drain the node.
+
+Run the following:
+
+```bash
+ncn-mw# kubectl drain --ignore-daemonsets --delete-local-data $NODE
+```
+
+You may run into pods that cannot be gracefully evicted due to Pod Disruption Budgets (PDB), for example:
+
+```text
+error when evicting pod "<pod>" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
+```
+
+In this case, there are some options. If the service is scalable, you can increase the scale to start up another pod on another node, and then the drain will be able to delete it. However, it will probably be necessary to force the deletion of the pod:
+
+```bash
+ncn-mw# kubectl delete pod [-n <namespace>] --force --grace-period=0 <pod>
+```
+
+This will delete the offending pod, and Kubernetes should schedule a replacement on another node. You can then rerun the `kubectl drain` command, and it should report that the node is drained
```

Comment:
```suggestion
This will delete the offending pod, and Kubernetes should schedule a replacement on another node. Then rerun the `kubectl drain` command, and it should report that the node is drained.
```